test/spectests/spectests.cpp

```diff
@@ -0,0 +1,183 @@
+#include "execute.hpp"
+#include "parser.hpp"
+#include <nlohmann/json.hpp>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+
+namespace fs = std::filesystem;
+using json = nlohmann::json;
+
+namespace
+{
+constexpr auto JsonExtension = ".json";
+
+void run_tests_from_file(const fs::path& path)
+{
+    std::cout << "Running tests from " << path << "\n";
+
+    std::ifstream test_file{path};
+    json j;
+    test_file >> j;
+
+    fizzy::Instance instance;
+    for (const auto& cmd : j.at("commands"))
+    {
+        const auto type = cmd.at("type").get<std::string>();
+
+        std::cout << "Line " << cmd.at("line").get<int>() << ": " << std::flush;
+
+        if (type == "module")
+        {
+            const auto filename = cmd.at("filename").get<std::string>();
+            std::cout << "Instantiating " << filename << "\n";
+
+            std::ifstream wasm_file{fs::path{path}.replace_filename(filename)};
+
+            const auto wasm_binary = fizzy::bytes(
+                std::istreambuf_iterator<char>{wasm_file}, std::istreambuf_iterator<char>{});
+
+            try
+            {
+                instance = fizzy::instantiate(fizzy::parse(wasm_binary));
+            }
+            catch (fizzy::parser_error const&)
+            {
+                std::cout << "FAILED Parsing failed.\n";
+                continue;
```

Comment:
I would rethrow here, to stop the execution, because the following tests might not make sense in this case

Reply:
Better to skip the following tests till next module, I think that will be easy when we support instantiating more than one module for some tests.